pkg/utils/strategy_params.go

```diff
@@ -0,0 +1,61 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/sets"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"sigs.k8s.io/descheduler/pkg/api"
+)
+
+// ValidateAndParseStrategyParams is used to contain validated strategy params
+type ValidatedStrategyParameters struct {
+	ThresholdPriority  int32
+	IncludedNamespaces sets.String
+	ExcludedNamespaces sets.String
+	LabelSelector      labels.Selector
+	NodeFit            bool
+}
+
+// ValidateAndParseStrategyParams will validate parameters to ensure that they do not contain invalid values.
+// If valid, it will return a ValidatedStrategyParameters struct.
+func ValidateAndParseStrategyParams(ctx context.Context, client clientset.Interface, params *api.StrategyParameters) (*ValidatedStrategyParameters, error) {
```

Comment:
Sorry, but I think I was unclear with how I meant the struct pattern could be applied to other strategies in https://github.com/kubernetes-sigs/descheduler/pull/576#discussion_r638758100

It will be difficult to extract this function to a util that can be used by other strategies, because many strategies have different params. In my +1, I just meant the pattern of returning a struct as opposed to multiple values could be applied to the other strategies.

With that in mind, let's drop the change to extract this function out. I do like the intent, so we could open an issue to explore it further. But it's not particularly relevant to this PR so we don't need it right now